src/app/pages/api/embedding.ts

```diff
@@ -0,0 +1,32 @@
+import dotenv from 'dotenv';
+import * as path from 'path';
+import { OpenaiEmbeddingType } from '@/app/types/OpenaiApiType';
+dotenv.config({ path: path.resolve(__dirname, '../../../..', '.env.local') });
+
+const createEmbedding = async (query: string): Promise<OpenaiEmbeddingType> => {
+  const embeddingResponse = await fetch(
+    'https://api.openai.com/v1/embeddings',
+    {
+      method: 'POST',
+      headers: {
+        Authorization: 'Bearer ' + process.env.OPENAI_API_KEY,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        model: 'text-embedding-ada-002',
+        input: query.replaceAll('\n', ' '),
+      }),
+    }
+  );
+
+  if (embeddingResponse.status !== 200) {
+    throw 'Failed to create embedding for question';
```

Comment:
can throw error status code + message as well in case of failure, this is just backend so we'll want to see what's going wrong